crates/bevy_pbr/src/meshlet/from_mesh.rs

```diff
@@ -19,80 +26,342 @@ impl MeshletMesh {
+            let groups = group_meshlets(
+                simplification_queue.clone(),
+                &connected_meshlets_per_meshlet,
+            );
+
+            let next_lod_start = meshlets.len();
+
+            for group_meshlets in groups.values().filter(|group| group.len() > 1) {
+                // Simplify the group to ~50% triangle count
+                let Some((simplified_group_indices, mut group_error)) =
+                    simplify_meshlet_groups(group_meshlets, &meshlets, &vertices, lod_level)
+                else {
+                    continue;
+                };
+
+                // Add the maximum child error to the parent error to make parent error cumulative from LOD 0
+                // (we're currently building the parent from its children)
+                group_error += group_meshlets.iter().fold(group_error, |acc, meshlet_id| {
+                    acc.max(bounding_spheres[*meshlet_id].self_lod.radius)
+                });
+
+                // Build a new LOD bounding sphere for the simplified group as a whole
+                let mut group_bounding_sphere = convert_meshlet_bounds(compute_cluster_bounds(
+                    &simplified_group_indices,
+                    &vertices,
+                ));
+                group_bounding_sphere.radius = group_error;
+
+                // For each meshlet in the group set their parent LOD bounding sphere to that of the simplified group
+                for meshlet_id in group_meshlets {
+                    bounding_spheres[*meshlet_id].parent_lod = group_bounding_sphere;
+                }
+
+                // Build new meshlets using the simplified group
+                let new_meshlets_count = split_simplified_groups_into_new_meshlets(
+                    &simplified_group_indices,
+                    &vertices,
+                    &mut meshlets,
+                );
+
+                // Calculate the culling bounding sphere for the new meshlets and set their LOD bounding spheres
+                let new_meshlet_ids = (meshlets.len() - new_meshlets_count)..meshlets.len();
+                bounding_spheres.extend(
+                    new_meshlet_ids
+                        .map(|meshlet_id| {
+                            compute_meshlet_bounds(meshlets.get(meshlet_id), &vertices)
+                        })
+                        .map(convert_meshlet_bounds)
+                        .map(|bounding_sphere| MeshletBoundingSpheres {
+                            self_culling: bounding_sphere,
+                            self_lod: group_bounding_sphere,
+                            parent_lod: MeshletBoundingSphere {
+                                center: group_bounding_sphere.center,
+                                radius: f32::MAX,
+                            },
+                        }),
+                );
+            }
+
+            simplification_queue = next_lod_start..meshlets.len();
```

Comment:
One thing I don't understand about this loop is why each iteration processes the whole range from current lod to max lod - why doesn't it just process for the current lod from the previous lod?

Reply:
I had misunderstood. I get it now.